### PR TITLE
Define _GNU_SOURCE for GNU builds

### PR DESCRIPTION
**Pull request by tbrowder**
The definition results in quelling the compile warnings for libuv on Debian Buster with gcc

The changed MoarVM source files have #ifndef/#endif guards added to prevent redefinition warnings

### CODE REVIEW

**Review by niner — CHANGES_REQUESTED** [review]
The change itself looks good, but the commit message does not explain why the change is needed. You do explain that in the PR but that will not make it into the git history and be lost to the inquiring developer.

File: Configure.pl [hunk 1]
```diff
@@ -430,6 +430,9 @@ sub uniq {
 push @cflags, '-DMVM_HEAPSNAPSHOT_FORMAT=' . $config{heapsnapformat};
 push @cflags, $ENV{CFLAGS} if $ENV{CFLAGS};
 push @cflags, $ENV{CPPFLAGS} if $ENV{CPPFLAGS};
+
+push @cflags, '-D_GNU_SOURCE'; # mainly for libuv
```
Comment (MasterDuke17) [inline Configure.pl:434]: What about making this `push @cflags, '-D_GNU_SOURCE' unless $args{'has-libuv'};`?

Comment (tbrowder) [reply Configure.pl:434]: Okay, will do.

**Comment by tbrowder** [comment]
> the commit message does not explain why the change is needed

I made a more complete explanatory comment in the last commit.

**Review by niner — APPROVED** [review]

**Review by MasterDuke17 — APPROVED** [review]

**Pull request merged**